src/service/resources-impl.js

```diff
@@ -1052,9 +1053,13 @@ export class Resources {
           // an element's boundary is not changed above the viewport after
           // resize.
           resize = true;
-        } else if (bottomDisplacedBoundary <= viewportRect.top + topOffset) {
+        } else if (scrollTop > 1 &&
+            bottomDisplacedBoundary <= viewportRect.top + topOffset) {
           // 5. Elements above the viewport can only be resized if we are able
-          // to compensate the height change by setting scrollTop.
+          // to compensate the height change by setting scrollTop and only if
```

Comment:
The effect can be seen on http://amp.autoplaced.com/ if you enable the experiment: AMP.toggleExperiment('amp-auto-ads').

It's not easy to capture in an example page, as amp-auto-ads requires a remote configuration, which seems hard to mock out in an example page.